### PR TITLE
Autotracker: Basic zooming and moves with velocity estimation

### PR DESCRIPTION
**Pull request by hawkeye217**
This PR adds a few features to the autotracker:

1) Basic zooming - There is plenty of room for improvement, but this implementation uses ONVIF absolute zooming to conservatively zoom in and out on an autotracked object. Relative zooming is also supported with some Dahua and Amcrest PTZs (which is what I had to test). Relative zooming will zoom concurrently with a pan/tilt movement.

2) Movements based on Norfair's estimation of the object's velocity. Faster moving objects now cause more PTZ movement so that the object is not out of the frame after a move. This requires the user to "calibrate" their camera on startup so that Frigate can measure movement times of the user's camera. These values are then stored and a simple linear regression is calculated to predict the time of a pan/tilt move. The intercept and coefficients of the calculation are written back to the config file.

I've tested with my setup and it's working, but I'd welcome anyone else with a PTZ to help test and improve the feature!

### CODE REVIEW

**Comment by NickM-27** [comment]
Is there a reason why a user would want to use absolute zoom if relative zoom is available?

**Comment by NickM-27** [comment]
Thanks, I think we'll want some kind of detailed explanation on the docs for that. 

I also think instead of two separate fields for zoom it would be preferable to have one field with an enum of `none`, `absolute`, and `relative`

File: docs/docs/configuration/autotracking.md [hunk 1]
```diff
@@ -66,15 +72,31 @@ cameras:
         return_preset: home
         # Optional: Seconds to delay before returning to preset. (default: shown below)
         timeout: 10
+        # Optional: Values generated automatically by a camera calibration. Do not modify these manually. (default: None)
+        movement_weights: None
```
Comment (NickM-27) [inline docs/docs/configuration/autotracking.md:76]:
```suggestion
        movement_weights: []
```

File: docs/docs/configuration/index.md [hunk 2]
```diff
@@ -586,9 +592,11 @@ cameras:
         required_zones:
           - zone_name
         # Required: Name of ONVIF preset in camera's firmware to return to when tracking is over. (default: shown below)
-        return_preset: preset_name
+        return_preset: home
         # Optional: Seconds to delay before returning to preset. (default: shown below)
         timeout: 10
+        # Optional: Values generated automatically by a camera calibration. Do not modify these manually. (default: None)
+        movement_weights: None
```
Comment (NickM-27) [inline docs/docs/configuration/index.md:599]:
```suggestion
        movement_weights: []
```

File: frigate/config.py [hunk 3]
```diff
@@ -160,6 +163,27 @@ class PtzAutotrackConfig(FrigateBaseModel):
     timeout: int = Field(
         default=10, title="Seconds to delay before returning to preset."
     )
+    movement_weights: Optional[Union[float, List[float]]] = Field(
+        default=[],
+        title="Internal value used for PTZ movements based on the speed of your camera's motor.",
+    )
+
+    @validator("movement_weights", pre=True)
```
Comment (NickM-27) [inline frigate/config.py:171]: maybe we should add a test in the config tests for this

**Review by NickM-27 — CHANGES_REQUESTED** [review]

File: frigate/track/norfair_tracker.py [hunk 4]
```diff
@@ -273,9 +273,11 @@ def match_and_update(self, frame_time, detections):
                 min(self.detect_config.width - 1, estimate[2]),
                 min(self.detect_config.height - 1, estimate[3]),
             )
+            estimate_velocity = tuple(t.estimate_velocity.flatten().astype(int))
```
Comment (NickM-27) [inline frigate/track/norfair_tracker.py:276]: should we only calculate this if this camera has autotracking enabled?

Comment (hawkeye217) [reply frigate/track/norfair_tracker.py:276]: This is just pulling an existing field already present on every Norfair detection. The rest of the frigate codebase doesn't use it, so it shouldn't cause any problems just to have in the `obj` object.

Comment (NickM-27) [reply frigate/track/norfair_tracker.py:276]: Oh okay cool, wasn't sure if retrieving that value started the calcuation

**Review by NickM-27 — APPROVED** [review]

File: frigate/track/norfair_tracker.py (hunk 4 above)
Comment (hawkeye217) [reply frigate/track/norfair_tracker.py:276]: For reference: https://tryolabs.github.io/norfair/2.2/reference/tracker/#norfair.tracker.TrackedObject.estimate_velocity

**Pull request merged**